src/object/obj_tx.c

```diff
@@ -799,26 +1452,39 @@ dc_tx_commit_trigger(tse_task_t *task, struct dc_tx *tx)
 		D_GOTO(out, rc);
 
 	crt_req_addref(req);
-	tcca = crt_req_get(req);
-	D_ASSERT(tcca != NULL);
-
-	tcca->tcca_tx = tx;
-	tcca->tcca_req = req;
+	tcca.tcca_req = req;
+	tcca.tcca_tx = tx;
+	tcca.tcca_args = args;
 
 	rc = tse_task_register_comp_cb(task, dc_tx_commit_cb,
-				       tcca, sizeof(*tcca));
+				       &tcca, sizeof(tcca));
 	if (rc != 0) {
 		/* drop ref from crt_req_addref. */
 		crt_req_decref(req);
-		D_ERROR("Failed to register completion cb: rc = %d\n", rc);
+		D_ERROR("Failed to register completion cb: "DF_RC"\n",
+			DP_RC(rc));
 
 		D_GOTO(out, rc);
 	}
 
-	/* FIXME:
-	 * 1. Prepare bulk.
-	 * 2. Pack sub requests into the RPC.
-	 */
+	oci = crt_req_get(req);
+	D_ASSERT(oci != NULL);
+
+	rc = dc_cont_hdl2uuid(tx->tx_coh, &oci->oci_co_hdl, &oci->oci_co_uuid);
+	D_ASSERT(rc == 0);
+
+	uuid_copy(oci->oci_pool_uuid, tx->tx_pool->dp_pool);
+	oci->oci_map_ver = tx->tx_pm_ver;
+	oci->oci_flags = DRF_CPD_LEADER | (tx->tx_retry ? ORF_RESEND : 0);
```

Comment:
Why isn't this based on DTF_RETRY_COMMIT?

Reply:
The commit RPC for restarted TX will also contains "ORF_RESEND".  But for such case, "DTF_RETRY_COMMIT" is not set.